addon-test-support/@ember/test-helpers/-internal/helper-hooks.ts

```diff
@@ -0,0 +1,49 @@
+type Hook = (...args: any[]) => void | Promise<void>;
+type HookUnregister = {
+  unregister: () => void;
+};
+
+const registeredHooks = new Map<string, Set<Hook>>();
+
+/**
+ * Registers a hook function to be run during the invocation of a test helper.
+ *
+ * @public
+ * @param {string} helperName The name of the test helper in which to run the hook.
+ * @param {Function} hook The hook function to run when the test helper is invoked.
+ * @returns {HookUnregister} An object containing an unregister function that will unregister
+ *                           the specific hook registered to the helper.
+ */
+export function registerHook(helperName: string, hook: Hook): HookUnregister {
```

Comment:
`helperName` doesn't seem acccurate. It's used like `$helperName:$start_or_end` currently.
I'm wondering if we should split it to 2 args, like:

```ts
export function registerHook(helperName: string, checkpoint: string, hook: Hook): HookUnregister
```

or maybe even separate utils for `start` and `end`, like:
```ts
export function beforeEach(helperName: string, hook: Hook): HookUnregister
export function afterEach(helperName: string, hook: Hook): HookUnregister
```

or just rename the arg?

Reply:
I think your first suggestion actually makes sense. This API evolved a bit along the way, it used to just be the raw helper name. I think splitting these into `helperName` and `when` (`checkpoint`, or whatever) sounds reasonable.

Reply:
I added an `order` argument (I think order makes the most sense). I'm also going to validate the string passed in, to ensure people don't pass invalid `order` options (though I believe it won't error).